src/io/output_vtk.cc

```diff
@@ -159,11 +160,9 @@ int OutputVTK::write_data(void)
 
 void OutputVTK::make_subdirectory()
 {
-    string main_file="./" + this->_base_filename; // guarantee that find_last_of succeeds
-    OLD_ASSERT( main_file.substr( main_file.size() - 4) == ".pvd" , "none");
-    unsigned int last_sep_pos=main_file.find_last_of(DIR_DELIMITER);
-    main_output_dir_=main_file.substr(2, last_sep_pos-2);
-    main_output_basename_=main_file.substr(last_sep_pos+1);
+	OLD_ASSERT( string(this->_base_filename).substr( string(this->_base_filename).size() - 4) == ".pvd" , "none");
+	main_output_dir_ = this->_base_filename.parent_path();
+	main_output_basename_ = this->_base_filename.filename();
     main_output_basename_=main_output_basename_.substr(0, main_output_basename_.size() - 4); // 5 = ".pvd".size() +1
 
     FilePath fp(main_output_dir_ + DIR_DELIMITER + main_output_basename_ + DIR_DELIMITER + "__tmp__", FilePath::output_file);
```

Comment:
Better avoid usage of  DIR_DELIMITER at all, introducing a constructor accepting vector of strings that will be connected by correct delimiter.